services/RecipesAPI/RecipesAPI/RecipesAPI/Services/ReviewService.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RecipesAPI.Database;
+using RecipesAPI.Entities.Reviews;
+using RecipesAPI.Model.Common;
+using RecipesAPI.Model.Reviews.Add;
+using RecipesAPI.Model.Reviews.Get;
+using RecipesAPI.Services.Interfaces;
+
+namespace RecipesAPI.Services
+{
+    public class ReviewService : IReviewService
+    {
+        private readonly RecipeDbContext _context;
+
+        public ReviewService(RecipeDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<PaginatedResult<IEnumerable<GetReviewDTO>>> GetReviewsByRecipeId(Guid recipeId, CancellationToken ct, int pageNumber, int pageSize)
+        {
+
+            if (pageNumber <= 0) pageNumber = 1;
+            if (pageSize <= 0) pageSize = 10;
+
+            var query = _context.Reviews
```

Comment:
still different than in other queries, also the Select... r.ReviewPhotos.Select(p => p...).ToList() materializes the query early (afaik)